### PR TITLE
Feat: custom sort for processes

### PR DESCRIPTION
**Pull request by Stef-Rousset**
#### :tophat: Description
Allow custom sort of participatory processes depending on environment variable (this is a backport of a specific feature from decidim-tou)

#### :pushpin: Related Issues
*Link your PR to an issue*

- [Odoo card](https://opensourcepolitics.odoo.com/web#id=2578&cids=1&menu_id=325&action=471&model=project.task&view_type=form)

#### Testing
*Describe the best way to test or validate your PR.*

1. As an admin, go to an assembly, add Related participatory processes and update.
2. As a user, go to the show page of the assembly and see that Related participatory processes are not sorted (cf screenshot 1)
3. Stop your server and relaunch it with `SORT_PROCESSES_BY_DATE=true bundle exec rails s`
4. As a user, go to the show page of the assembly and see that Related participatory processes are sorted with (cf screenshot 2):
- active first (sorted by end date)
- future second (sorted by start date)
- past third (sorted by end date reverse)

#### Tasks
- [ X] Add specs
- [ X] Add note about overrides in OVERLOADS.md

### :camera: Screenshots
ONE =>
one

TWO =>
two

### CODE REVIEW

File: config/secrets.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@ default: &default
       throttle:
         max_requests: <%= ENV["THROTTLING_MAX_REQUESTS"]&.to_i || 100 %>
         period: <%= ENV["THROTTLING_PERIOD"]&.to_i || 60 %>
+    participatory_processes:
+      sort_by_date: <%= ENV["SORT_PROCESSES_BY_DATE"] || false %>
```
Comment (Quentinchampenois) [inline config/secrets.yml:47]: Env variable have type String, you can cast it to TrueClass or FalseClass by comparing value with `"true"`

```suggestion
      sort_by_date: <%= ENV.fetch("SORT_PROCESSES_BY_DATE", "false") == "true" %>
```

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Just a suggestion about the value of env variable

File: config/secrets.yml (hunk 1 above)
Comment (Stef-Rousset) [reply config/secrets.yml:47]: ok, thx for your suggestion @Quentinchampenois, I am going to apply it !

**Review by BarbaraOliveira13 — APPROVED** [review]
I ran it, works 🎉 Very nice job! 💪

**Review by luciegrau — APPROVED** [review]

**Review by Quentinchampenois — APPROVED** [review]

**Pull request merged**